src/persist-client/src/cli/admin.rs

```diff
@@ -664,3 +665,138 @@ async fn force_gc(
 
     Ok(Box::new(machine))
 }
+
+/// WIP
+pub const CATALOG_FORCE_COMPACTION_FUEL: Config<usize> = Config::new(
+    "persist_catalog_force_compaction_fuel",
+    // WIP
+    1024,
+    "WIP",
+);
+
+/// WIP
+pub const CATALOG_FORCE_COMPACTION_WAIT: Config<Duration> = Config::new(
+    "persist_catalog_force_compaction_duration",
+    // WIP
+    Duration::from_secs(1),
+    "WIP",
+);
+
+/// Attempts to compact all batches in a shard into a minimal number.
+///
+/// This destroys most hope of filter pushdown doing anything useful. If you
+/// think you want this, you almost certainly want something else, please come
+/// talk to the persist team before using.
+///
+/// This is accomplished by adding artificial fuel (representing some count of
+/// updates) to the shard's internal Spine of batches structure on some schedule
+/// and doing any compaction work that results. Both the amount of fuel and
+/// cadence are dynamically tunable. However, it is not clear exactly how much
+/// fuel is safe to add in each call: 10 definitely seems fine, usize::MAX may
+/// not be. This also adds to the danger in "dangerous".
+///
+/// This is intentionally not even hooked up to `persistcli admin` for the above
+/// reasons.
+pub async fn dangerous_force_compaction_and_break_pushdown<K, V, T, D>(
```

Comment:
So right now this just loops forever, but typing that up, it felt a bit awkward to me. Thinking now that it should exit when batches are < either 2 or 1 and if the user wants to keep the shard continually as compacted as possible, then they're always free to call it in a loop.

Reply:
I ended up really liking the option of calling this sync, partially because it also enables calling it repeatedly in a task if that's what you want, so both behaviors are desirable. Done